frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -517,14 +524,19 @@ export function doHealthPing(
     connect()
   }
 
-  // Make sure we don't retry faster than timeoutMs. This is required because
-  // in some cases things fail very quickly, and all our fast retries end up
-  // bogging down the browser.
   const retry = (errorNode: React.ReactNode): void => {
-    const tryDuration = (Date.now() - tryTimestamp) / 1000
-    const retryTimeout = tryDuration < timeoutMs ? timeoutMs - tryDuration : 0
-
-    retryCallback(totalTries, errorNode)
+    // Adjust retry time by +- 20% to spread out load
+    const jitter = Math.random() * 0.4 - 0.2
+    // Exponential backoff to reduce load from health pings when experiencing
+    // persistent failure. Starts at minimumTimeoutMs.
+    const timeoutMs =
+      totalTries === 1
+        ? minimumTimeoutMs
+        : minimumTimeoutMs * 2 ** (totalTries - 1) * (1 + jitter)
+    // timeoutMs === 0 ? minimumTimeoutMs : timeoutMs * 2 * (1 + jitter)
```

Comment:
Is this commented line intended to be here?

Reply:
Nope. I'll remove it before I merge